app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -19,89 +17,101 @@ import { removeNotification } from '../../actions/notifications';
   }
 
-  render() {
-    if (!this.props.projectBoard.isFetched) {
-      return <ProjectLoading data-id="project-loading" />;
+  const columns = [
+    {
+      title: I18n.t("projects.show.chilly_bin"),
+      renderAction: () =>
+        <AddStoryButton
+          onAdd={() => createStory({
+            state: Story.status.UNSCHEDULED
+          })}
+        />,
+      children: <Stories stories={chillyBinStories} />
+    },
+    {
+      title: `${I18n.t("projects.show.backlog")} / ${I18n.t("projects.show.in_progress")}`,
+      renderAction: () =>
+        <AddStoryButton
+          onAdd={() => createStory({
+            state: Story.status.UNSTARTED
+          })}
+        />,
+      children: <Sprints sprints={backlogSprints} />
+    },
+    {
+      title: I18n.t("projects.show.done"),
+      children: <Sprints sprints={doneSprints} fetchStories={fetchPastStories} />
     }
+  ];
 
-    const { 
-      projectId, 
-      createStory, 
-      closeHistory,
-      notifications, 
-      removeNotification, 
-      history,
-      projectBoard
-    } = this.props;
+  return (
+    <div className="ProjectBoard">
+      <StorySearch projectId={projectId} loading={projectBoard.search.loading} />
 
-    return (
-      <div className="ProjectBoard">
-        <StorySearch projectId={projectId} loading={projectBoard.search.loading} />
+      <SideBar />
 
-        <Notifications
-          notifications={notifications}
-          onRemove={removeNotification}
-        />
+      <Notifications
+        notifications={notifications}
+        onRemove={removeNotification}
+      />
 
-        <Column title={I18n.t("projects.show.chilly_bin")}
-          renderAction={() =>
-            <AddStoryButton
-              onAdd={() => createStory({
-                state: Story.status.UNSCHEDULED
-              })}
+      {
+        projectBoard.reverse
+          ? 
+            <Columns
+              data-id="reversed-column"
+              columns={columns.reverse()}
             />
-          }
-        >
-          <Stories stories={this.props.chillyBinStories} />
-        </Column>
+          : 
+            <Columns
+              data-id="normal-column"
+              columns={columns}
+            />
+      }
 
-        <Column
-          title={`${I18n.t("projects.show.backlog")} /
-          ${I18n.t("projects.show.in_progress")}`}
-          renderAction={() =>
-            <AddStoryButton
-              onAdd={() => createStory({
-                state: Story.status.UNSTARTED
-              })}
-            />}
-        >
-          <Sprints
-            sprints={this.props.backlogSprints}
-          />
-        </Column>
+      <SearchResults />
 
+      {
+        history.status !== 'DISABLED' &&
         <Column
-          title={I18n.t("projects.show.done")}
+          onClose={closeHistory}
+          title={[I18n.t("projects.show.history"), "'", history.storyTitle, "'"].join(' ')}
```

Comment:
Do you think we can improve it somehow?

Reply:
Yep, I believe than with template string will be better.